CRM/Import/Parser.php

```diff
@@ -646,6 +656,28 @@ protected function doPostImportActions() {
     UserJob::update(FALSE)->addWhere('id', '=', $userJob['id'])->setValues(['metadata' => $this->userJob['metadata']])->execute();
   }
 
+  public function queue() {
+    $dataSource = $this->getDataSourceObject();
+    $totalRowCount = $totalRows = $dataSource->getRowCount(['new']);
+    $queue = Civi::queue('user_job_' . $this->getUserJobID(), ['type' => 'Sql', 'error' => 'abort']);
+    $offset = 0;
+    $batchSize = 5;
```

Comment:
@eileenmcnaughton I assume the small batch size (`5`) is to facilitate inspection (*make it easy to watch the batching mechanism*)? There's a separate/follow-up issue to extract this to a setting? (That's fine - not a blocker.)

But if we wind up with a constant-value in the stable-release -- it should probably be a bit higher (like... 20 or 100 or 500 or 1000...)

Reply:
@totten yeah - I think the batch size will be an option on the first page of the import

Reply:
I would probably go for around 50 as a default - that should import in 30 seconds on most sites

Reply:
Seems like there may be sweet spot for it. 5 I get for testing but doing imports now and it makes it quite slow. I can try witching that hardcoded limit around and see.

Reply:
Thanks - let us know what you find

Reply:
I played around with [some guesstimates about what determines the max batch limit](https://gist.github.com/totten/8f18e04c2198f098521946d6ca3a3222). For example, one might assume that:

* Each PHP worker is limited to 64mb RAM (the baseline required by Civi installer)
* Each HTTP request is bound by the default timeout specified by `apache.org` and `nginx.org` (ie 60s)
    * Per `php.net`, the default timeout for a PHP request is actually lower (30s), but most importers call `set_time_limit(0)`.
* Each import-task requires an average 100ms runtime and 100kb memory.

The gist is a bit fiddly, and (*while fiddling*) the projected limits ranged somewhere around 300 - 600.

For a hard-coded/application-level default, I'd probably vote for 100...